src/Display/Slider/Slider.tsx

```diff
@@ -10,234 +10,193 @@ import LeftArrow from './LeftArrow';
 import RightArrow from './RightArrow';
 import SliderItem from './SliderItem';
 
-class Slider extends React.Component<Props, State> {
-  static Item = SliderItem;
-
-  interval: ReturnType<typeof setTimeout>;
-  sliderContainerRef: React.RefObject<HTMLDivElement>;
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      translateValue: 0,
-      screenSize: 0,
-      index: 1,
-    };
-    this.sliderContainerRef = React.createRef();
+/**
+ * Custom react hook that utilizes an internal state to trigger re-renders when
+ * the exported state update function is called.
+ */
+const useForceUpdate = () => {
+  const [, setIndex] = React.useState(0);
+  return () => setIndex(index => index + 1);
+};
+
+const Slider = ({
+  afterChange,
+  children,
+  autoplay,
+  initialItem,
+  index: controlledIndex,
+  className,
+  fullContent,
+  arrowWhite,
+  removeDots,
+  containerRef,
+}: Props) => {
+  const interval = React.useRef<ReturnType<typeof setTimeout>>();
+  const privateContainerRef = React.useRef<HTMLDivElement>();
+  const sliderContainerRef = containerRef || privateContainerRef;
+
+  const childrenCount = React.Children.toArray(children).filter(
+    child => !isNil(child)
+  ).length;
+
+  if (initialItem && (initialItem < 0 || initialItem >= childrenCount)) {
+    throw new Error(
+      `initialItem ${initialItem} is out of bounds (min 0, max ${childrenCount -
+        1})`
+    );
   }
 
-  previousSlide = () => {
-    const { afterChange } = this.props;
-    const { index, translateValue, screenSize } = this.state;
+  if (
+    controlledIndex &&
+    (controlledIndex < 0 || controlledIndex >= childrenCount)
+  ) {
+    throw new Error(
+      `controlledIndex ${controlledIndex} is out of bounds (min 0, max ${childrenCount -
+        1})`
+    );
+  }
 
-    if (index !== 1) {
-      this.setState({
-        index: index - 1,
-        translateValue: translateValue + screenSize,
-      });
+  const [internalIndex, setInternalIndex] = React.useState<number>(
+    initialItem || 0
+  );
 
-      if (afterChange !== undefined) {
-        afterChange(index - 1);
-      }
-    }
-  };
+  const index = isNil(controlledIndex) ? internalIndex : controlledIndex;
```

Comment:
I think we can have an `useEffect` for the controlled index, and I guess it will be easier to understand logics between controlled and uncontrolled indexes.
```
useEffect(() => {
  setInternalIndex(controlledIndex);
}, [controlledIndex] )
```

Reply:
but I think the real problem is the original implementation initialized many internal states only at `componentDidMount`, so when `children` or `initialItem` changes, the component might not work anymore.